x-pack/plugins/maps/public/kibana_services.ts

```diff
@@ -22,6 +24,16 @@ export function setStartServices(core: CoreStart, plugins: MapsPluginStartDepend
   coreStart = core;
   pluginsStart = plugins;
 }
+
+let IndexNameFormComponent: React.ComponentType<IndexNameFormProps>;
+let FileUploadComponent: React.ComponentType<FileUploadComponentProps>;
+export async function setAsyncComponents() {
```

Comment:
Not sure this is the best place for this. This will cause the file_upload plugin to load even if the users never opens the upload or draw UI. Also, there is no guarantee that the file_upload plugin will return before the wizard is opened. This could cause errors if you try to render before the file plugin is loaded.

Suggestions would be:
1) keep the current behavior and load when used.
2) Expose a component from file upload that loads the plugin boundary internally so maps is not exposed to this implementation detail. The component would internally async load the file upload plugin. Before the plugin is loaded, it would return some sort of loading UI view. After the plugin is loaded, it would return the component.